lib/fs.js

```diff
@@ -89,30 +88,7 @@ function copyObject(source) {
 }
 
 function rethrow() {
-  // TODO(thefourtheye) Throw error instead of warning in major version > 7
-  process.emitWarning(
-    'Calling an asynchronous function without callback is deprecated.',
-    'DeprecationWarning', 'DEP0013', rethrow
-  );
-
-  // Only enable in debug mode. A backtrace uses ~1000 bytes of heap space and
-  // is fairly slow to generate.
-  if (DEBUG) {
-    var backtrace = new Error();
-    return function(err) {
-      if (err) {
-        backtrace.stack = err.name + ': ' + err.message +
-                          backtrace.stack.substr(backtrace.name.length);
-        throw backtrace;
-      }
-    };
-  }
-
-  return function(err) {
-    if (err) {
-      throw err;  // Forgot a callback but don't know where? Use NODE_DEBUG=fs
-    }
-  };
+  throw new TypeError('"callback" argument must be a function');
```

Comment:
Wherever this error ends up being thrown, please include a comment that indicates that this previously was a deprecation with ID code `DEP0013`